compat/src/portals.js

```diff
@@ -1,78 +1,76 @@
 import { createElement, hydrate, render, __u as _unmount } from 'preact';
 
-class ContextProvider {
-	getChildContext() {
-		return this.props.context;
-	}
-	render(props) {
-		return props.children;
-	}
+function ContextProvider(props) {
+	this.getChildContext = () => props.context;
+	return props.children;
 }
 
 /**
  * Portal component
  * @param {object | null | undefined} props
+ *
+ * TODO: this could use the "fake root node" trick from the partial hydration demo
  */
 function Portal(props) {
-	let _this = this;
```

Comment:
What is the reason for dropping this? Would save us 2 bytes keeping the `let _this = this` and using `_this` afterwards

Reply:
yeah I was 50/50 on this. I think some more refactoring can get the total down, but the `_this` alias was masking some of the other possible changes. I was mostly just trying to switch out arrow functions that had the possibility to cause accidental closures for `this` access.